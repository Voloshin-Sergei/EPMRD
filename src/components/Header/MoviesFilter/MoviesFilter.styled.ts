import styled from 'styled-components';
import { colors } from '../../../styles/variables/colors.styled';
import { breakpoints } from '../../../styles/variables/responsive.styled';

export const StyledFilter = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem 5rem;
  background-color: ${colors.colorPrimaryDarkLight};

  @media screen and (max-width: ${breakpoints.mobileSm}) {
    flex-direction: column;
  }
`;

export const StyledMoviesCount = styled.p`
  color: ${colors.colorPrimaryDark};
  font-size: 1.8rem;
`;

export const StyledButtonsContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;
export const StyledButtonsContainerDesc = styled.p`
  margin-right: 1rem;
  font-size: 1.8rem;
`;

export const StyledList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style: none;
`;
export const StyledListItem = styled.li`
  margin-right: 1rem;
`;
