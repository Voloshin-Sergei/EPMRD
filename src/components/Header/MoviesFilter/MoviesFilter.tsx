import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Button } from 'Components/common/Button';
import { setFilter } from 'Store/actions/searchMovieAction';
import { RootState } from 'Store/reducers';
import { FilterTag } from '../../Header/Header';
import {
  StyledFilter,
  StyledMoviesCount,
  StyledButtonsContainer,
  StyledButtonsContainerDesc,
  StyledList,
  StyledListItem,
} from './MoviesFilter.styled';

export interface MoviesFilterProps {
  filterTags: FilterTag[];
}

export const MoviesFilter: React.FC<MoviesFilterProps> = ({ filterTags }) => {
  const { filter, movies } = useSelector((state: RootState) => state.searchMovieReducer);

  const dispatch = useDispatch();

  const handleFilterClick = (activeFilter: string) => () => {
    dispatch(setFilter(activeFilter));
  };

  return (
    <StyledFilter>
      <StyledMoviesCount>{movies.length} movies found</StyledMoviesCount>
      <StyledButtonsContainer>
        <StyledButtonsContainerDesc>Sort by</StyledButtonsContainerDesc>
        <StyledList>
          {filterTags.map((tag: FilterTag) => (
            <StyledListItem key={tag.label}>
              <Button
                onClick={handleFilterClick?.(tag.type)}
                typeBtn="filterTagBtn"
                variant="secondary"
                dataTestId="filter-tag-btn"
                active={tag.type === filter}
              >
                {tag.label}
              </Button>
            </StyledListItem>
          ))}
        </StyledList>
      </StyledButtonsContainer>
    </StyledFilter>
  );
};
