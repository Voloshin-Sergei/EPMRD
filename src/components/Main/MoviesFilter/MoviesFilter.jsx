import React from 'react';
import style from './MoviesFilter.module.scss';

export const MoviesFilter = () => (
  <div className={style.sort}>
    <div>
      <p className={style.result}>7 movies found</p>
    </div>
    <div className={style.buttons}>
      <p className={style.text}>Sort by</p>
      <ul className={style.list}>
        <li className={style.item}>
          <button className={style.button} type="button">
            release
          </button>
        </li>
        <li className={style.item}>
          <button className={style.button} type="button">
            date
          </button>
        </li>
        <li className={style.item}>
          <button className={style.button} type="button">
            rating
          </button>
        </li>
      </ul>
    </div>
  </div>
);
