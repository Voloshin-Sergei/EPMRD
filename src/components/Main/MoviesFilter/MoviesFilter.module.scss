@import 'Styles/variables/colors';
@import 'Styles/variables/responsive';

.filter {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem 5rem;
  background-color: darken($color-primary, 10%);

  @media screen and (max-width: $mobile--sm) {
    flex-direction: column;
  }
}

.result {
  font-size: 1.8rem;
  color: $color-primary--dark;
}

.buttons {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.text {
  font-size: 1.8rem;
  margin-right: 1rem;
}

.list {
  display: flex;
  flex-wrap: wrap;
  list-style: none;
}

.item {
  margin-right: 1rem;
}

.tag {
  font-size: 1.6rem;
  transform: scale(0.9);

  &:hover {
    transform: scale(1);
  }

  &:focus {
    color: $color-secondary;
    font-size: 1.8rem;
  }
}
