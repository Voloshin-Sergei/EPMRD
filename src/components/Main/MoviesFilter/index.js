export { MoviesFilter as default } from './MoviesFilter';
