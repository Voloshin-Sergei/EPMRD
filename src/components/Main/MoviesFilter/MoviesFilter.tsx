import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Button } from 'Components/common/Button';
import { setFilterCategory, searchMovie } from 'Store/actions/searchMovieAction';
import { RootState } from 'Store/reducers';
import { FilterTag } from '../Main';

import style from './MoviesFilter.module.scss';

export interface MoviesFilterProps {
  filterTags: Array<FilterTag>;
}

export const MoviesFilter: React.FC<MoviesFilterProps> = ({ filterTags }) => {
  const { movies, searchValue, searchCategory, sortCategory } = useSelector(
    (state: RootState) => state.searchMovieReducer,
  );

  const dispatch = useDispatch();

  // useEffect(() => {
  //   dispatch(searchMovie(sortCategory, searchValue, searchCategory));
  // }, [sortCategory]);

  const moviesSortBy = (category: string): void => {
    dispatch(setFilterCategory(category));
  };

  return (
    <div className={style.filter}>
      <div>
        <p className={style.result}>{movies.length} movies found</p>
      </div>
      <div className={style.buttons}>
        <p className={style.text}>Sort by</p>
        <ul className={style.list}>
          {filterTags.map((tag: FilterTag) => (
            <li key={tag.label} className={style.item}>
              <Button
                callback={moviesSortBy}
                className={style.tag}
                variant="secondary"
                dataTestId="filter-tag-btn"
                typeCategory={tag.type}
                activeClassName={tag.type === sortCategory ? style.active : ''}
              >
                {tag.label}
              </Button>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};
