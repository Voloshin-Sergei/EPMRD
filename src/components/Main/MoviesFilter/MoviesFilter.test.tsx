import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { MoviesFilter}  from './MoviesFilter';
Enzyme.configure({ adapter: new Adapter() });

const component = render(<MoviesFilter/>);

describe("MoviesFilter component", () => {
    it('should render MoviesFilter component', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render 3 buttons', () => {
    const buttons = component.find("button")
    expect(buttons).toHaveLength(3);
  });
});