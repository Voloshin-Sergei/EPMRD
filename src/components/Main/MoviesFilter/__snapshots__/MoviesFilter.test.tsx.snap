// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesFilter component should render MoviesFilter component 1`] = `
<div
  class="filter"
>
  <div>
    <p
      class="result"
    >
      7 movies found
    </p>
  </div>
  <div
    class="buttons"
  >
    <p
      class="text"
    >
      Sort by
    </p>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <button
          class="button tag secondary"
          type="button"
        >
          release
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="button tag secondary"
          type="button"
        >
          date
        </button>
      </li>
      <li
        class="item"
      >
        <button
          class="button tag secondary"
          type="button"
        >
          rating
        </button>
      </li>
    </ul>
  </div>
</div>
`;
